### DISCUSSION TITLE
Google Lighthouse Pagespeed

### DISCUSSION THREAD

**Discussion by azheng003** [gatsbyjs/gatsby#31754] [Help]
I'm working on a site and I'm having trouble improving the pagespeed scores. I used the webpack analyzer and tried removing a few big modules but that didn't make a huge difference. Is there something glaring that I'm missing? I'm not quite sure what a good webpack analyzer result looks like. Should I be concerned about the modules outside of `commons.js`? Thanks in advance.

![scores](https://user-images.githubusercontent.com/49768461/120728060-ecac9480-c490-11eb-90bd-c54f01ed83ea.png)
![webpack](https://user-images.githubusercontent.com/49768461/120727983-bf5fe680-c490-11eb-97f4-6b30a3a626a9.png)

```
Opportunities
These suggestions can help your page load faster. They don't directly affect the Performance score.

Reduce unused JavaScript
1.03 s
Reduce initial server response time
0.56 s

Diagnostics
More information about the performance of your application. These numbers don't directly affect the Performance score.

Image elements do not have explicit width and height
Avoid enormous network payloads Total size was 8,277 KiB
Serve static assets with an efficient cache policy 21 resources found
Minimize main-thread work 6.4 s
Reduce JavaScript execution time 3.7 s
Avoid an excessive DOM size 950 elements
Avoid chaining critical requests 9 chains found
Keep request counts low and transfer sizes small 75 requests • 16,621 KiB
Largest Contentful Paint element 1 element found
Avoid large layout shifts 4 elements found
Avoid long main-thread tasks 13 long tasks found
```

**Comment by decimoseptimo** [comment]
Have you checked your packages against bundlephobia?
I've fixed bad scores in the past by removing packages that introduced huge boileplate html, css and js code.

Create a branch and start removing stuff, commit, and lighthouse test.

**Comment by herecydev** [comment]
I agree with @decimoseptimo , from my experience what you have is far too much but it sort of depends on what you're trying to accomplish. Also, make sure to get your site behind a CDN or something similar, HTTP2, compression, etc.

That will give you the most realistic result in lighthouse.

**Comment by azheng003** [comment]
Removing a bunch of packages help my score a little bit, but changing the way my fonts were loaded and pre-loading the fonts helped a lot. There's still more progress to be made though.
